src/py42/_internal/clients/alerts.py

```diff
@@ -73,19 +79,45 @@ def get_all_rules(self, sort_key=u"CreatedAt", sort_direction=u"DESC"):
             self._get_alert_rules,
             u"ruleMetadata",
             tenant_id=tenant_id,
+            groups=None,
             sort_key=sort_key,
             sort_direction=sort_direction,
         )
 
-    def get_rules_by_name(self, rule_name):
-        rule_pages = self.get_all_rules()
-        matched_rules = []
-        for rule_page in rule_pages:
-            rules = rule_page[u"ruleMetadata"]
-            for rule in rules:
-                if rule_name.lower() == rule[u"name"].lower():
-                    matched_rules.append(rule)
-        return matched_rules
+    def get_all_rules_by_name(self, rule_name, sort_key=u"CreatedAt", sort_direction=u"DESC"):
+        tenant_id = self._user_context.get_current_tenant_id()
+        return get_all_pages(
+            self._get_alert_rules,
+            u"ruleMetadata",
+            tenant_id=tenant_id,
+            groups=[
```

Comment:
Should we create a shared function to create this blob for these functions? It could take the term as a parameter. We could also look into using code from the query services too, so that at any time, it is mindless to create FilterClause/Group looking JSON

Reply:
we already have one! switched to use that.